tests/ffi/agent-flood/test.sh

```diff
@@ -64,15 +64,43 @@ run_test_containers(){
     done
 }
 
+# Extract the Network mode from qm quadlet file
+get_qm_network_mode(){
+    qm_config_file="/usr/share/containers/systemd/qm.container"
+
+    # Check if the configuration file exists
+    if [ ! -f "$qm_config_file" ]; then
+        echo "Configuration file not found: $qm_config_file"
+        exit 1
+    fi
+
+    # Extract the Network using awk
+    qm_config_file="/usr/share/containers/systemd/qm.container"
+    qm_network_mode=$(awk -F'=' '/Network=private/ { print $2 }' "$qm_config_file")
+    echo "${qm_network_mode}"
+}
+
 disk_cleanup
 prepare_test
 reload_config
 prepare_images
 
+# Assign value to ${controller_host_ip} according to qm network mode
+controller_host_ip="localhost"
+if [ "$(get_qm_network_mode)" == "private" ]; then
+    controller_host_ip=$(hostname -I | awk '{print $1}')
+fi
+
 #Stop QM bluechi-agent
 exec_cmd "podman exec -it qm /bin/bash -c \
          \"systemctl stop bluechi-agent\""
 
+# This is the workaround when ControllerHost is not in /etc/qm/bluechi/agent.conf.d/agent.conf
+# Add ControllerHost to /etc/qm/bluechi/agent.conf.d/agent.conf
+if test -f "/etc/qm/bluechi/agent.conf.d/agent.conf"; then
```

Comment:
In case file does not exist. 
It should return an error, qm bc-agent will not connect

Reply:
done